### DISCUSSION TITLE
Bluetooth mesh

### DISCUSSION THREAD

**Discussion by Peter-Jackson22** [rbaron/b-parasite#107] [Q&A]
Apologies if this is obvious to everyone else, but I'm much handier with a soldering iron than software.

I noticed that the bluetooth chips mentioned mesh capabilities, is this something that needs a central processor to enable or can it link up between b-parasites?

I work in agricultural research and was thinking about deploying ~100 of them across a field over a prolonged period to measure soil moisture every 30 minutes or so. Connecting to each individually would be a pain. (sorry if I'm misunderstanding the technology)

looking as a replacement for these: https://i-button.co.uk/humidity-ds1923

**Comment by rbaron** [comment]
That's a good question and an interesting use case.

The SDK does support a few different meshy protocols: [Bluetooth mesh](https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/protocols/bt_mesh/index.html) (built on top of Bluetooth Low Energy), [Zigbee](https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/protocols/zigbee/index.html), [Thread](https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/protocols/thread/index.html). As far as I know, and simplifying a lot, all these mesh technologies require special nodes to "relay" messages inside the network. These relay nodes essentially are on all the time and are usually mains powered (think a light bulb or a smart speaker). The end devices (like sensors), frequently battery powered, are then allowed to sleep most of the time.

b-parasite on the other hand uses Bluetooth Low Energy in a much simpler way. It is only an oblivious, tiny-battery-power end device. It spends most of its time sleeping and then just broadcasts its values once every 10 minutes and go right back to sleep. It doesn't even know if there's some device listening to it. In this sense, b-parasite really is designed for residential, short distance installations. We can extend it to WiFi ranges if we use some [ESP32 proxies](https://github.com/rbaron/b-parasite/blob/main/code/nrf-connect/samples/ble/README.md#bthome-encoding), but not to arbitrary distances.

For long distance outdoor use, b-parasite running Bluetooth Low Energy is not going to work. Even if we implemented a mesh topology (which we don't currently, except for an experimental/uncertified/educational [Zigbee example](https://github.com/rbaron/b-parasite/blob/main/code/nrf-connect/samples/zigbee/README.md)), it would be challenging if you don't have access to mains power (for relay nodes), or if you want to avoid single points of failure. 

For long distance, outdoor use, I would look into [LoRa](https://en.wikipedia.org/wiki/LoRa)/LoRaWan. It's a different radio technology that favors long distance over throughput. People seem to get kilometers of range in open areas, and it's low energy enough to run off batteries (although probably not as tiny as a CR2032).

**Reply by Peter-Jackson22** [reply]
Thanks for clearing this up for me. Between this and some other reading I've found, it looks like I was wildly optimistic about the bluetooth ranges (I was looking more at the theoretical maximums).

with your suggestion of LoRa, I've already managed to find some appropriate sensors for work. (although I'm still going to do a small production run of the parasite for myself, JLCPCB have E73-2G4M08S1EX in stock, is this an acceptable switch for E73-2G4M08S1E??)

**Reply by rbaron** [reply]
I have never used anything other than the C variant, but people have had success with the CX and E variants. You'd need to check the datasheet of the EX and compare with the E.
